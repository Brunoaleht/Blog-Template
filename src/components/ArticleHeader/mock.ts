import { ArticleHeaderProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  title: 'ReactHook Studies',
  slug: 'react-hook-studies',
  cover: {
    coverData: {
      attributes: {
        alternativeText: null,
        url: 'https://res.cloudinary.com/dxldesniw/image/upload/v1677769491/thumbnail_MKS_entrevista_87872be63c_60bb358adf.jpg',
      },
    },
  },
  excerpt:
    'I made this project using various react hooks, such as useEffect to make a call to the api, use State to keep my data in state, useContext and useReducer to do the work in the general context in the shopping cart.',
  author: {
    authorData: {
      id: '1',
      attributes: {
        displayName: 'Bruno Alexandre',
        slug: 'bruno-alexandre',
      },
    },
  },
  categories: {
    categoryData: [
      {
        id: '2',
        attributes: {
          displayName: 'ReactJs',
          slug: 'react-js',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'JavaScript',
          slug: 'java-script',
        },
      },
    ],
  },
} as ArticleHeaderProps;
