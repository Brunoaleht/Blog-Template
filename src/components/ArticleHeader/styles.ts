import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';
export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.large};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    ${HeadingStyle} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;
export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.fonts.sizes.medium};
  `}
`;
export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
