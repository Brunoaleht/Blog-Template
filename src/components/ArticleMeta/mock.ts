import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    authorData: {
      id: '1',
      attributes: {
        displayName: 'Bruno Alexandre',
        slug: 'bruno-alexandre',
      },
    },
  },
  categories: {
    categoryData: [
      {
        id: '2',
        attributes: {
          displayName: 'ReactJs',
          slug: 'react-js',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'JavaScript',
          slug: 'java-script',
        },
      },
    ],
  },
} as ArticleMetaProps;
