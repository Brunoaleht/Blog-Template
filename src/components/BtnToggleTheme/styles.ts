import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 2;
  `}
`;
export const Label = styled.label`
  ${() => css`
    position: relative;
    display: inline-block;
    width: 5rem;
    height: 2.5rem;
    line-height: 0;
    overflow: hidden;
    font-size: 0;
    color: rgba(0, 0, 0, 0);
  `}
`;
export const BtnInput = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;
    &:checked + ${Slider} {
      background: ${theme.colors.secondary};
    }
    &:focus + ${Slider} {
      border: solid 1px ${theme.colors.secondary};
    }
    &:checked + ${Slider}::before {
      transform: translateX(1.95rem);
      background: ${theme.colors.primary};
    }
  `}
`;
export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    cursor: pointer;
    background: ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;
    border: solid 1px ${theme.colors.darkerGray};
    border-radius: 2rem;
    &::before {
      content: '';
      position: absolute;
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.1rem;
      background: ${theme.colors.textDark};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
