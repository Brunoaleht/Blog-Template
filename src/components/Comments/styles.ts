import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
