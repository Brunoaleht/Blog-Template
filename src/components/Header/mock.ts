import { HeaderProps } from '.';
export default {
  blogName: 'Bruno Alexandre',
  blogDescription:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Possimus doloremque nulla eius aperiam modi error facere, commodi.',
  logo: {
    logoData: {
      attributes: {
        alternativeText: null,
        url: 'https://res.cloudinary.com/dxldesniw/image/upload/v1677869914/logo1_6dca30fa4b.png',
      },
    },
  },
  showText: true,
} as HeaderProps;
