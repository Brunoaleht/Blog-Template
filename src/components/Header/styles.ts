import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding-bottom: ${theme.spacings.huge};
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.textDark};
    font-size: ${theme.fonts.sizes.small};
    margin: 0 auto;
    ${HeadingStyle} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }
    @media ${theme.media.lteSmall} {
      display: flex;
      flex-flow: column wrap;
      & > ${HeadingStyle} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }
    }
  `}
`;
export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
    p {
      margin-top: 0;
    }
    @media ${theme.media.lteSmall} {
      margin-left: 0;
    }
  `}
`;
