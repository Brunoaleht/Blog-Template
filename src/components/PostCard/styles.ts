import styled, { css } from 'styled-components';

import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background-color: ${theme.colors.white};
    box-shadow: 0 0 10px rgb(0, 0, 0, 0.2);
    border-radius: 8px;
    padding: 1rem;
    cursor: pointer;
    transition: transform 100ms ease-in-out;
    ${HeadingStyles} {
      margin: ${theme.spacings.xsmall} 0 ${theme.spacings.small} 0;
    }
    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }
    &:hover {
      transform: scale(1.05);
    }
    &:hover a,
    &:hover img {
      color: ${theme.colors.secondary};
      opacity: 0.8;
    }
  `}
`;
export const Cover = styled.img`
  ${() => css`
    width: 100%;
    transition: opacity 300ms ease-in-out;
  `}
`;
export const Excerpt = styled.p`
  ${() => css`
    margin-top: 0;
  `}
`;
