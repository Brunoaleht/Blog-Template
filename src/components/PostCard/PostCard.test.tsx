import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostCard, PostCardProps } from '.';
import mock from './mock';

const props: PostCardProps = mock;

describe('<PostCard />', () => {
  it('should render', () => {
    renderTheme(<PostCard {...props} />);

    expect(screen.getByRole('heading', { name: props.attributes.title })).toBeInTheDocument();
    expect(screen.getByRole('img')).toHaveAttribute('src', props.attributes.cover.coverData.attributes.url);
    expect(screen.getAllByRole('link')[0]).toHaveAttribute('href', `/post/${props.attributes.slug}`);
    expect(screen.getByText(props.attributes.excerpt)).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<PostCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
