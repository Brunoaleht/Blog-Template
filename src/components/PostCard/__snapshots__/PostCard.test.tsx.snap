// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
.c3 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #ffffff;
  box-shadow: 0 0 10px rgb(0,0,0,0.2);
  border-radius: 8px;
  padding: 1rem;
  cursor: pointer;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  -webkit-transition: transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
}

.c0 .c2 {
  margin: .8rem 0 1.6rem 0;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c0:hover a,
.c0:hover img {
  color: #dc143c;
  opacity: 0.8;
}

.c1 {
  width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c4 {
  margin-top: 0;
}

<div>
  <div
    class="c0"
    style="max-width: 32rem;"
  >
    <a
      href="/post/react-hook-studies"
    >
      <img
        class="c1"
        src="https://res.cloudinary.com/dxldesniw/image/upload/v1677769491/thumbnail_MKS_entrevista_87872be63c_60bb358adf.jpg"
      />
    </a>
    <h2
      class="c2 c3"
    >
      <a
        href="/post/react-hook-studies"
      >
        ReactHook Studies
      </a>
    </h2>
    <p
      class="c4"
    >
      I made this project using various react hooks, such as useEffect to make a call to the api, use State to keep my data in state, useContext and useReducer to do the work in the general context in the shopping cart.
    </p>
  </div>
</div>
`;
